tests/gocase/util/slot.go

```diff
@@ -0,0 +1,852 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package util
+
+func GetKeyWithSlotNum(slotNum int) string {
+	return slotTable[slotNum]
+}
+
+// A table of the shortest possible alphanumeric string that is mapped by
+// redis's crc16 to any given redis cluster slot.
+//
+// The array indexes are slot numbers, so that given a desired slot, this string is guaranteed
+// to make redis cluster route a request to the shard holding this slot
+var slotTable = [...]string{
```

Comment:
Could some comments be added to explain how this table was generated?

Reply:
I don't know how it's generated also. Ditto the original commit on Kvrocks or Redis: https://github.com/redis/redis/blob/d96f47cf06b1cc24b82109e0e87ac5428517525a/src/crc16_slottable.h

I guess it's random trying and filter..

Reply:
🤣